src/settings/controllers/CustomStatusController.js

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2019 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import SettingController from "./SettingController";
+import dis from "../../dispatcher";
+
+export default class CustomStatusController extends SettingController {
+    onChange(level, roomId, newValue) {
+        // Dispatch setting change so that some components that are still visible when the
+        // Settings page is open (such as RoomTiles) can reflect the change.
+        dis.dispatch({
+            action: "feature_custom_status_changed",
+            value: newValue,
```

Comment:
the value alone does not mean much without the applicable `level` and `roomId` which changed. It doesn't look like the `value` is actually used anywhere, so it should be safe to take out rather than passing around 3 extraneous values.

Also just a head's up that `onChange` can be called for ultimately inconsequential changes to the setting. If the user were to update the setting at a higher priority level such that the lower priority values don't matter, and then updates the lower priority levels, there will be no observable change in the setting's value despite `onChange` being called.

Reply:
Ah, okay, I'll remove the value then. Thanks for the extra info!